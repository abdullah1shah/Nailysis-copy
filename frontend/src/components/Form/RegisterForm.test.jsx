import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { vi } from 'vitest';
import RegisterForm from './RegisterForm';

// Mock dependencies
vi.mock('../SocialLogin/SocialLogin', () => ({
  default: () => <div data-testid="social-login">Social Login Component</div>
}));

vi.mock('react-router-dom', async () => {
  const actual = await vi.importActual('react-router-dom');
  return {
    ...actual,
    useNavigate: () => vi.fn()
  };
});

vi.mock('../../api', () => ({
  default: {
    post: vi.fn()
  }
}));

vi.mock('react-toastify', () => ({
  toast: {
    warning: vi.fn(),
    success: vi.fn(),
    error: vi.fn()
  }
}));

// Wrapper component for Router
const renderWithRouter = (component) => {
  return render(
    <BrowserRouter>
      {component}
    </BrowserRouter>
  );
};

describe('RegisterForm', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  test('renders the registration form with all required fields', () => {
    renderWithRouter(<RegisterForm route="/api/register" />);
    
    expect(screen.getByText('Create Your Account')).toBeInTheDocument();
    expect(screen.getByLabelText(/first name/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/last name/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/email/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/phone/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/date of birth/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/address/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/emergency contact/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/^password$/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/confirm password/i)).toBeInTheDocument();
  });

  test('renders gender radio buttons', () => {
    renderWithRouter(<RegisterForm route="/api/register" />);
    
    expect(screen.getByRole('radio', { name: /^male$/i })).toBeInTheDocument();
    expect(screen.getByRole('radio', { name: /^female$/i })).toBeInTheDocument();
    expect(screen.getByRole('radio', { name: /^other$/i })).toBeInTheDocument();
    expect(screen.getByRole('radio', { name: /^prefer not to say$/i })).toBeInTheDocument();
  });

  test('selects gender radio button', () => {
    renderWithRouter(<RegisterForm route="/api/register" />);
    
    const maleRadio = screen.getByRole('radio', { name: /^male$/i });
    fireEvent.click(maleRadio);
    
    expect(maleRadio).toBeChecked();
  });

  test('renders submit button and social login', () => {
    renderWithRouter(<RegisterForm route="/api/register" />);
    
    expect(screen.getByRole('button', { name: /submit/i })).toBeInTheDocument();
    expect(screen.getByTestId('social-login')).toBeInTheDocument();
  });

  test('updates first name input value', () => {
    renderWithRouter(<RegisterForm route="/api/register" />);
    
    const firstNameInput = screen.getByLabelText(/first name/i);
    fireEvent.change(firstNameInput, { target: { value: 'John' } });
    
    expect(firstNameInput.value).toBe('John');
  });

  test('updates last name input value', () => {
    renderWithRouter(<RegisterForm route="/api/register" />);
    
    const lastNameInput = screen.getByLabelText(/last name/i);
    fireEvent.change(lastNameInput, { target: { value: 'Doe' } });
    
    expect(lastNameInput.value).toBe('Doe');
  });

  test('updates email input value', () => {
    renderWithRouter(<RegisterForm route="/api/register" />);
    
    const emailInput = screen.getByLabelText(/email/i);
    fireEvent.change(emailInput, { target: { value: 'john@example.com' } });
    
    expect(emailInput.value).toBe('john@example.com');
  });

  test('updates phone input value', () => {
    renderWithRouter(<RegisterForm route="/api/register" />);
    
    const phoneInput = screen.getByLabelText(/phone/i);
    fireEvent.change(phoneInput, { target: { value: '1234567890' } });
    
    expect(phoneInput.value).toBe('1234567890');
  });

  test('updates date of birth input value', () => {
    renderWithRouter(<RegisterForm route="/api/register" />);
    
    const dobInput = screen.getByLabelText(/date of birth/i);
    fireEvent.change(dobInput, { target: { value: '1990-01-01' } });
    
    expect(dobInput.value).toBe('1990-01-01');
  });

  test('updates address textarea value', () => {
    renderWithRouter(<RegisterForm route="/api/register" />);
    
    const addressInput = screen.getByLabelText(/address/i);
    fireEvent.change(addressInput, { target: { value: '123 Main St' } });
    
    expect(addressInput.value).toBe('123 Main St');
  });

  test('updates emergency contact input value', () => {
    renderWithRouter(<RegisterForm route="/api/register" />);
    
    const emergencyInput = screen.getByLabelText(/emergency contact/i);
    fireEvent.change(emergencyInput, { target: { value: '0987654321' } });
    
    expect(emergencyInput.value).toBe('0987654321');
  });

  test('updates password input value', () => {
    renderWithRouter(<RegisterForm route="/api/register" />);
    
    const passwordInput = screen.getByLabelText(/^password$/i);
    fireEvent.change(passwordInput, { target: { value: 'password123' } });
    
    expect(passwordInput.value).toBe('password123');
  });

  test('updates confirm password input value', () => {
    renderWithRouter(<RegisterForm route="/api/register" />);
    
    const confirmPasswordInput = screen.getByLabelText(/confirm password/i);
    fireEvent.change(confirmPasswordInput, { target: { value: 'password123' } });
    
    expect(confirmPasswordInput.value).toBe('password123');
  });

  test('form has correct attributes', () => {
    renderWithRouter(<RegisterForm route="/api/register" />);
    
    const form = screen.getByRole('form');
    expect(form).toHaveClass('form-container');
  });

  test('all input fields are required', () => {
    renderWithRouter(<RegisterForm route="/api/register" />);
    
    expect(screen.getByLabelText(/first name/i)).toBeRequired();
    expect(screen.getByLabelText(/last name/i)).toBeRequired();
    expect(screen.getByLabelText(/email/i)).toBeRequired();
    expect(screen.getByLabelText(/phone/i)).toBeRequired();
    expect(screen.getByLabelText(/date of birth/i)).toBeRequired();
    expect(screen.getByLabelText(/address/i)).toBeRequired();
    expect(screen.getByLabelText(/emergency contact/i)).toBeRequired();
    expect(screen.getByLabelText(/^password$/i)).toBeRequired();
    expect(screen.getByLabelText(/confirm password/i)).toBeRequired();
  });

  test('gender radio buttons have correct values', () => {
    renderWithRouter(<RegisterForm route="/api/register" />);
    
    const maleRadio = screen.getByLabelText(/^male$/i);
    const femaleRadio = screen.getByLabelText(/^female$/i);
    const otherRadio = screen.getByLabelText(/^other$/i);
    const preferNotToSayRadio = screen.getByLabelText(/^prefer not to say$/i);
    
    expect(maleRadio).toHaveAttribute('value', 'M');
    expect(femaleRadio).toHaveAttribute('value', 'F');
    expect(otherRadio).toHaveAttribute('value', 'O');
    expect(preferNotToSayRadio).toHaveAttribute('value', 'P');
  });

  test('only one gender can be selected at a time', () => {
    renderWithRouter(<RegisterForm route="/api/register" />);
    
    const maleRadio = screen.getByLabelText(/^male$/i);
    const femaleRadio = screen.getByLabelText(/^female$/i);
    
    fireEvent.click(maleRadio);
    expect(maleRadio).toBeChecked();
    expect(femaleRadio).not.toBeChecked();
    
    fireEvent.click(femaleRadio);
    expect(femaleRadio).toBeChecked();
    expect(maleRadio).not.toBeChecked();
  });
});

