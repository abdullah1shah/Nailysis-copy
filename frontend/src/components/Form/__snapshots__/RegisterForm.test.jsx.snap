// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RegisterForm Snapshots matches snapshot 1`] = `
<div
  class="form"
>
  <section
    class="main scrollablediv"
  >
    <h2>
      Create Your Account
    </h2>
    <form
      class="form-container"
    >
      <div
        class="inputRow"
      >
        <div
          class="inputGroup halfWidth"
        >
          <label
            for="first_name"
          >
            First Name
          </label>
          <input
            id="first_name"
            placeholder="Enter your first name"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="inputGroup halfWidth"
        >
          <label
            for="last_name"
          >
            Last Name
          </label>
          <input
            id="last_name"
            placeholder="Enter your last name"
            required=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="inputGroup"
      >
        <label
          for="email"
        >
          Email
        </label>
        <input
          id="email"
          placeholder="Enter your email"
          required=""
          type="email"
          value=""
        />
      </div>
      <div
        class="inputGroup"
      >
        <label
          for="phone"
        >
          Phone
        </label>
        <input
          id="phone"
          placeholder="Enter your phone number"
          required=""
          type="tel"
          value=""
        />
      </div>
      <div
        class="inputGroup"
      >
        <label
          for="date_of_birth"
        >
          Date of Birth
        </label>
        <input
          id="date_of_birth"
          required=""
          type="date"
          value=""
        />
      </div>
      <div
        class="inputGroup"
      >
        <label>
          Gender
        </label>
        <div
          class="radioGroup"
        >
          <label
            for="male"
          >
            <input
              id="male"
              name="gender"
              required=""
              type="radio"
              value="M"
            />
            Male
          </label>
          <label
            for="female"
          >
            <input
              id="female"
              name="gender"
              required=""
              type="radio"
              value="F"
            />
            Female
          </label>
          <label
            for="other"
          >
            <input
              id="other"
              name="gender"
              required=""
              type="radio"
              value="O"
            />
            Other
          </label>
          <label
            for="preferNotToSay"
          >
            <input
              id="preferNotToSay"
              name="gender"
              required=""
              type="radio"
              value="P"
            />
            Prefer Not to Say
          </label>
        </div>
      </div>
      <div
        class="inputGroup"
      >
        <label
          for="address"
        >
          Address
        </label>
        <textarea
          id="address"
          placeholder="Enter your full address"
          required=""
          style="all: unset; display: block; width: 98%; padding: 0.5rem; border: 1px solid #ccc; border-radius: 4px; font-size: 1rem; line-height: 1.5; background-color: rgb(255, 255, 255); color: rgb(0, 0, 0); overflow-y: none;"
        />
      </div>
      <div
        class="inputGroup"
      >
        <label
          for="emergency_contact"
        >
          Emergency Contact
        </label>
        <input
          id="emergency_contact"
          placeholder="Enter emergency contact number"
          required=""
          type="tel"
          value=""
        />
      </div>
      <div
        class="inputGroup"
      >
        <label
          for="password"
        >
          Password
        </label>
        <input
          id="password"
          placeholder="Enter your password"
          required=""
          type="password"
          value=""
        />
      </div>
      <div
        class="inputGroup"
      >
        <label
          for="confirm_password"
        >
          Confirm Password
        </label>
        <input
          id="confirm_password"
          placeholder="Confirm your password"
          required=""
          type="password"
          value=""
        />
      </div>
      <button
        class="submitButton"
        type="submit"
      >
        Submit
      </button>
    </form>
    <p>
      or
    </p>
    <div
      data-testid="social-login"
    >
      Social Login Component
    </div>
  </section>
</div>
`;
